troposphere/awslambda.py

```diff
@@ -4,11 +4,35 @@
 
 class Code(AWSProperty):
     props = {
-        'S3Bucket': (basestring, True),
-        'S3Key': (basestring, True),
+        'S3Bucket': (basestring, False),
+        'S3Key': (basestring, False),
         'S3ObjectVersion': (basestring, False),
+        'ZipFile': (basestring, False)
     }
 
+    def __init__(self, title=None, **kwargs):
+        super(Code, self).__init__(title, **kwargs)
+
+        if 'ZipFile' in kwargs:
```

Comment:
Cool - it may be that we do it like this somewhere else, but I think validate() is probably the right place in general.  @markpeek - any opinion?